### PR TITLE
fix: tql parser hang on abnormal input

### PR DESCRIPTION
**Pull request by waynexia**
I hereby agree to the terms of the [GreptimeDB CLA](https://github.com/GreptimeTeam/.github/blob/main/CLA.md).

## What's changed and what's your intention?

I find this occasionally by a typo

image

And flamegraph tells me there is a dead loop 
image

## Checklist

- [ ] I have written the necessary rustdoc comments.
- [ ] I have added the necessary unit tests and integration tests.
- [ ] This PR requires documentation updates.

### CODE REVIEW

File: src/sql/src/parsers/tql_parser.rs [hunk 1]
```diff
@@ -123,22 +125,17 @@ impl<'a> ParserContext<'a> {
...
-        {
-            n += 1;
+    pub fn is_delimiter_token(token: &Token, delimiter_token: &[Token]) -> bool {
+        for delimiter in delimiter_token {
+            if delimiter == token {
+                return true;
+            }
         }
-        parser.peek_nth_token(n).token
+        false
```
Comment (evenyag) [inline src/sql/src/parsers/tql_parser.rs:134]:
```
delimiter_token.contains(token)
```

**Review by evenyag — APPROVED** [review]
Rest LGTM

**Review by WenyXu — APPROVED** [review]

File: src/sql/src/parsers/tql_parser.rs [hunk 2]
```diff
@@ -123,22 +125,17 @@ impl<'a> ParserContext<'a> {
         Ok(TqlParameters::new(start, end, step, lookback, query))
     }
 
-    fn find_next_delimiter_token(parser: &mut Parser) -> Token {
-        let mut n: usize = 0;
-        while !(Self::is_comma(&parser.peek_nth_token(n).token)
-            || Self::is_rparen(&parser.peek_nth_token(n).token))
-        {
-            n += 1;
+    pub fn is_delimiter_token(token: &Token, delimiter_token: &[Token]) -> bool {
```
Comment (etolbakov) [inline src/sql/src/parsers/tql_parser.rs:128]: nit: should it be `delimiter_tokens`?

**Comment by etolbakov** [comment]
sorry about that 😢 @waynexia

File: src/sql/src/parsers/tql_parser.rs (hunk 2 above)
Comment (waynexia) [reply src/sql/src/parsers/tql_parser.rs:128]: 👍 nice catch

@evenyag suggests replacing this function with`slice::contains`

**Comment by waynexia** [comment]
> sorry about that 😢 @waynexia

It's ok -- I won't consider this either without this typo

**Pull request merged**